mir-ci/mir_ci/robot_libraries/WaylandHid.py

```diff
@@ -48,20 +52,33 @@ async def release_pointer_button(self, button: str) -> None:
         await self.connect()
         self.button(Button[button], False)
 
+    @keyword
+    async def release_buttons(self) -> None:
+        """Release all buttons on the virtual pointer."""
+        await self.connect()
+        for button in Button:
+            self.button(button, False)
+
     @keyword
     async def get_absolute_from_proportional(self, x: float, y: float) -> Dict[str, float]:
         """Get the absolute position for the given output size proportions."""
+        await self.connect()
         assert 0 <= x <= 1, "x not in range 0..1"
         assert 0 <= y <= 1, "y not in range 0..1"
         assert self.output_width > 0, "Output width must be greater than 0"
         assert self.output_height > 0, "Output height must be greater than 0"
-        await self.connect()
         return {"x": x * self.output_width, "y": y * self.output_height}
 
+    @keyword
+    async def get_pointer_position(self) -> Dict[str, float]:
+        """Get the current pointer position."""
+        return self.pointer_position
+
     async def connect(self):
         """Connect to the display."""
         if not self.pointer:
             await super().connect()
+            await self.move_pointer_to_absolute(self.pointer_position["x"], self.pointer_position["y"])
```

Comment:
Shouldn't we actually move to `(0, 0)` and only set the current position here?

I mean, the position is undefined until now.

Reply:
I think `pointer_position` is defined as `(0,0)` in `__init__`.